File: Karan-Example-Project-Solutions/Java/Numerical/ChangeReturn.java
import java.util.Scanner;
import java.lang.Math;

public class ChangeReturn {

  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    System.out.println("What is the cost of the item: ");
    double cost = input.nextDouble();
    System.out.println("What is the amount paid with: ");
    double given = input.nextDouble();

    if (cost > given) {
      System.out.println("Not enough money given!");
    } else if (cost == given) {
      System.out.println("Exact change given!");
    } else {
      change(given - cost);
    }

    input.close();
  }

  private static void change(double n) {
    int total = (int)(n * 100);

    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;

    while (total >= 5) {
      if (total - 25 >= 0) {
        quarters++;
        total -= 25;
      } else if (total - 10 >= 0) {
        dimes++;
        total -= 10;
      } else if (total - 5 >= 0) {
        nickels++;
        total -= 5;
      }
    }
    pennies = total;
    System.out.println(quarters + " Quarters, " + dimes + " Dimes, "
                        + nickels + " Nickels, " + pennies + " Pennies. Total: $"
                         + Math.floor(n * 100) / 100);
  }

}

File: Karan-Example-Project-Solutions/Python/Numerical/ChangeReturn.py
import math

def change(n):
    total = str(math.floor(n * 100)/100)
    n = int(n * 100)
    quarters = 0
    dimes = 0
    nickels = 0
    pennies = 0

    while (n >= 5):
        if (n - 25 >= 0):
            quarters = quarters + 1
            n = n - 25
        elif (n - 10 >= 0):
            dimes = dimes + 1
            n = n - 10
        elif (n - 5 >= 0):
            nickels = nickels + 1
            n = n - 5
    pennies = n
    print(str(quarters), "Quarters,", str(dimes), "Dimes,",
          str(nickels), "Nickels,", str(pennies), "Pennies. Total:", total)

cost = float(input("What is the cost of the item: "))
given = float(input("What is the amount paid with: "))

if (cost > given):
    print("Not enough money given!")
elif (cost == given):
    print("Exact change given!")
else:
    change(given - cost)
